docs/DeveloperGuide.md

```diff
@@ -4,6 +4,382 @@ title: Developer Guide
+* Flashnotes keeps track of whether the user is currently inside a deck, and the name of the deck that the user is currently in.
+* The Parser will block these commands, taking arguments passed from Logic, which checks the state of Flashnotes through the model.
+
+#### Review related commands
+* The system only allows review-related commands (e.g. flip, correct, wrong, endReview)
+* The parser (logic) keeps track of whether the user is currently doing a review, and disables certain commands if the user is currently in a review session.
+
+#### Design considerations:
+* Alternative 1 (current choice): Checking of commands are done in the logic component.
+    * Pros: Model component does not need to keep track and handle invalid inputs by user
+    * Cons: Coupling between logic and model is increased
+* Alternative 2: Checking of commands are done in the model component.
+    * Pros; Reduced coupling
+    * Cons: Model has to handle commands, reducing cohesion.
+
+### Review Mode 
+Our FlashNotes application allows users to test their knowledge and mastery of flashcards through a review session.
+
+#### Implementation
+The review session is implemented by opening a new JavaFX window. This new window has its own command box (where users type in commands)
+and result display box (where the application displays messages to the user). On top of that, there is also the Individual Flashcard section
+of the window that shows the question of 1 flashcard. When the "flip" command is executed, 
+the answer to that question will be showed instead.
+
+To support the opening of this review session, the following commands were added:
+* `review` - A command that sets up the list of flashcards to review and opens a new review window displaying those cards.
+
+The following is a sequence diagram that demonstrates how a review command sets up the review session:
+![ReviewSequenceDiagram](images/ReviewSequenceDiagram.png)
+
+`ModelManager#shuffleReviewFlashcards` method sets up the list of flashcards to review inside `Model`.
+It duplicates the list of filtered flashcards in the model as of the moment that the review command was
+called, it then shuffles these cards using the `FXCollections.shuffle()` method, and it trims the list of
+flashcards to review according to the review limit set by users.
+
+#### Design considerations:
+* Alternative 1 (current choice): Open a new JavaFX window
+    * Pro: Differentiates the review mode from the other modes better visually, 
+    allows the UI for the review mode to be minimalistic, reducing distractions for users when reviewing their flashcards.
+    * Cons: We would have to create a new window with another command box and result display, and display
+    the main command box.
+* Alternative 2: Implement the review session in the same window as the rest of the application.
+    * Pro: Can use the same command box and result display so we would't have to create new command boxes and result
```

Comment:
should be would not, guide should be less verbal and more formal? (in language not tone)